src/main/sensors/gyro_init.c

```diff
@@ -645,6 +661,18 @@ bool gyroInit(void)
     // check if all enabled sensors are detected
     gyro.gyroEnabledBitmask = gyroDetectedFlags & gyroConfig()->gyro_enabled_bitmask;
 
+    // If no gyros are enabled but some are detected, enable at least the first detected gyro
+    // This prevents lockups when configuration is inconsistent
+    if (gyro.gyroEnabledBitmask == 0 && gyroDetectedFlags != 0) {
+        // Find the first detected gyro and enable it
+        for (int i = 0; i < GYRO_COUNT; i++) {
+            if (gyroDetectedFlags & GYRO_MASK(i)) {
+                gyro.gyroEnabledBitmask = GYRO_MASK(i);
+                break;
+            }
+        }
+    }
+
```

Comment:
IIRC some bittricks are already used in multigyro, so this may be ok:
```suggestion
    // use lowest set bit from detected flags
    gyro.gyroEnabledBitmask = gyroDetectedFlags & -gyroDetectedFlags;
```